Improve jwks error logging

cjrobbertse-ob opened the pull request:
- add 200 status code response check for get JWKS url
- add content-type logging and recommendations
- add unit tests for getJWKs errors
- add unit tests for getJWKs content type logging
- add unit tests for getJWKs status codes

cjrobbertse-ob committed:
Add JWKS content type validation and logging

Enhances getJwks to validate the Content-Type header against recommended and acceptable JWKS types, logging warnings or info as appropriate. Adds inArray helper and new tests for error handling and content type logging.

cjrobbertse-ob committed:
Improve JWKS status code error message and add test

Updated the error message in getJwks to include parentheses around status code and URL for clarity. Added a new test to verify error handling for different HTTP status codes returned by the JWKS endpoint.

cjrobbertse-ob committed:
Update JWKS content type log messages

Revised log messages in getJwks to clarify when acceptable or unexpected JWKS content types are found. Updated related test assertions to match new log message wording.

OBPeteS commented on pkg/authentication/trustanchor.go:
```diff
@@ -121,3 +144,14 @@ func parseCertificateChain(chain []string) ([]*x509.Certificate, error) {
+
+// inArray
+// check if a string is in an array of strings
+func inArray(target string, arr []string) bool {
```
This looks to be case-sensitive and content-type values are case-insensitive.  Suggest using something like strings.ToLower(target) and strings.ToLower(v)

OBPeteS commented on pkg/authentication/trustanchor_test.go:
```diff
@@ -80,3 +84,136 @@ func TestIsHSBCTrustAnchor(t *testing.T) {
+	}{
+		{
+			"200",
+			"https://keystore.openbanking.org.uk/0015800000jf9GgAAI/2QGUgXr5LAFcTUGkNP657c.jwks",
```
Use a non-production jwks: https://keystore.openbankingtest.org.uk/0015800001041RbAAI/0015800001041RbAAI.jwks

OBPeteS reviewed (changes requested):
Looks good, couple of changes requested

cjrobbertse-ob committed:
Update JWKS test URL in trustanchor tests

Replaces the JWKS URL in the 'valid' test case with a new test endpoint to ensure tests use a current and accessible resource.

Using non-production endpoint.

cjrobbertse-ob committed:
Make inArray string comparison case insensitive

Updated the inArray function to use strings.EqualFold for case-insensitive string comparison. This ensures that string matching works regardless of letter casing.

cjrobbertse-ob committed:
Add unit tests for inArray and JWKS functions

Introduces unit tests for the inArray function to verify case-insensitive string existence in arrays. Adds comments and improves test coverage for getJwks, getJwksContentType, and getJwksStatusCode functions.
